discovery.rst

```diff
@@ -84,23 +84,52 @@ Read more about setting up Puli in their `official documentation`_.
 Common Errors
 -------------
 
+Could not find resource using any discovery strategy
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you get an error saying "*Could not find resource using any discovery strategy.*" it means that all the
+discovery [#strategies]_ have failed. Most likely, your project is missing the message factories and/or a PRS-7
```

Comment:
define an anchor like this above the heading:

`.. _message-factory:`

and reference it with ``:ref:`linktext <message-factory>```. this works across the whole documentation, not just within a page. try to "namespace" the anchor a bit, just "strategies" would be a bit week. e.g. "discovery-strategies"